### PR TITLE
update cli for dev branch

### PR DESCRIPTION
**Pull request by rohit21agrawal**
consists of both the commits applied to 3.5.0-beta2 for updating the cli.

@alpaix @rrelyea @emgarten

### CODE REVIEW

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/RestoreCommand.cs [hunk 1]
```diff
@@ -117,9 +118,9 @@ internal static class RestoreCommand
 
                         if (inferRuntimes.HasValue())
                         {
-                            var defaultRuntimes = RequestRuntimeUtility.GetDefaultRestoreRuntimes(
-                                PlatformServices.Default.Runtime.OperatingSystem,
-                                PlatformServices.Default.Runtime.GetRuntimeOsName());
+                            string runtimeOSname = PlatformApis.GetRuntimeOsName();
```
Comment (alpaix) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/RestoreCommand.cs]: nit: use `var`

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Native/NativeMethods.Darwin.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
+﻿using System;
```
Comment (alpaix) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Native/NativeMethods.Darwin.cs]: header

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Native/NativeMethods.Darwin.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+using System.Runtime.InteropServices;
+
+namespace NuGet.CommandLine.XPlat.Native
```
Comment (alpaix) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Native/NativeMethods.Darwin.cs]: keep the root namespace `NuGet.CommandLine.XPlat`

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Native/NativeMethods.Unix.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Runtime.InteropServices;
+
+namespace NuGet.CommandLine.XPlat.Native
+{
+    internal static partial class NativeMethods
+    {
+        public static class Unix
+        {
+            public unsafe static string GetUname()
```
Comment (alpaix) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Native/NativeMethods.Unix.cs:17]: please document the method
-or-
use a meaningful name describing its purpose

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Native/PlatformApis.cs [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return Platform.Darwin;
+            }
+            return Platform.Unknown;
+        }
+#endif
+    }
+
+    public enum Platform
```
Comment (alpaix) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Native/PlatformApis.cs]: separate file

**Comment by alpaix** [comment]
:watch: As we're likely about to be stuck with borrowed native code we need to polish it a bit.

**Comment by emgarten** [comment]
I would expect the dependencies to be updated everywhere, not just for NuGet.Client.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Native/NativeMethods.Darwin.cs (hunk 3 above)
Comment (rohit21agrawal) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Native/NativeMethods.Darwin.cs]: @alpaix  shouldn't namespaces correspond to paths as far as possible?

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/RestoreCommand.cs (hunk 1 above)
Comment (rohit21agrawal) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/RestoreCommand.cs]: fixed

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Native/NativeMethods.Darwin.cs (hunk 2 above)
Comment (rohit21agrawal) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Native/NativeMethods.Darwin.cs]: fixed

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Native/NativeMethods.Unix.cs (hunk 4 above)
Comment (rohit21agrawal) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Native/NativeMethods.Unix.cs:17]: fixed

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Native/PlatformApis.cs (hunk 5 above)
Comment (rohit21agrawal) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Native/PlatformApis.cs]: fixed

**Comment by emgarten** [comment]
Talked about the dependencies offline with @rohit21agrawal I see the others were updated below. I suggest just double checking that there are no new restore warnings from this (Since warnings as errors isn't possible yet).

:shipit:

**Comment by alpaix** [comment]
:cool: :shipit:

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Native/NativeMethods.Darwin.cs (hunk 3 above)
Comment (alpaix) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Native/NativeMethods.Darwin.cs]: Actually we tend to limit namespace levels. See for instance NuGet.Protocol.Core.v3 classes.

Comment (rohit21agrawal) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Native/NativeMethods.Darwin.cs]: got it..fixed.

**Comment by rohit21agrawal** [comment]
Merged ... 
https://github.com/NuGet/NuGet.Client/commit/884d3313e40af8baedef675e3bd43d32717204e6

**Pull request closed without merging**